tests/acceptance/features/apiSharingNg/linkShare.feature

```diff
@@ -370,3 +370,151 @@ Feature: Create a share link for a resource
+          "hasPassword": {
+            "type": "boolean",
+            "enum": [true]
+          },
+          "id": {
+            "type": "string",
+            "pattern": "^[a-zA-Z]{15}$"
+          },
+          "link": {
+            "type": "object",
+            "required": [
+              "@libre.graph.displayName",
+              "@libre.graph.quickLink",
+              "preventsDownload",
+              "type",
+              "webUrl"
+            ],
+            "properties": {
+              "@libre.graph.displayName": {
+                "type": "string",
+                "enum": [""]
+              },
+              "@libre.graph.quickLink": {
+                "type": "boolean",
+                "enum": [false]
+              },
+              "preventsDownload": {
+                "type": "boolean",
+                "enum": [false]
+              },
+              "type": {
+                "type": "string",
+                "enum": ["<newRole>"]
+              },
+              "webUrl": {
+                "type": "string",
+                "pattern": "^%base_url%\/s\/[a-zA-Z]{15}$"
+              }
+            }
+          }
+        }
+      }
+      """
+    Examples:
+      | previousRole   | newRole        |
+      | view           | edit           |
+      | view           | blocksDownload |
+      | edit           | view           |
+      | edit           | blocksDownload |
+      | blocksDownload | edit           |
+      | blocksDownload | blocksDownload |
+
+
+  Scenario: update expiration date of a file's link share
+    Given user "Alice" has uploaded file with content "other data" to "textfile1.txt"
+    And user "Alice" has created the following link share:
+      | resourceType       | file                     |
+      | resource           | textfile1.txt            |
+      | space              | Personal                 |
+      | role               | view                     |
+      | password           | %public%                 |
+      | expirationDateTime | 2200-07-15T14:00:00.000Z |
+    When user "Alice" updates the last public link share using the Graph API with
+      | resourceType       | file                     |
+      | resource           | textfile1.txt            |
+      | space              | Personal                 |
+      | expirationDateTime | 2201-07-15T14:00:00.000Z |
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should match
+      """
+      {
+        "type": "object",
+        "required": [
+          "hasPassword",
+          "id",
+          "link",
+          "expirationDateTime"
+        ],
+        "properties": {
+          "hasPassword": {
+            "type": "boolean",
+            "enum": [true]
+          },
+          "id": {
+            "type": "string",
+            "pattern": "^[a-zA-Z]{15}$"
+          },
+          "expirationDateTime": {
+            "type": "string",
+            "enum": ["2201-07-15T23:59:59Z"]
```

Comment:
Is this the expected expiry date?
When the share invitation for a resource (file/folder) is sent with an expiry date `2200-07-15T14:00:00.000Z`, then the date in response is `2200-07-15T14:00:00Z`.

Reply:
yes its the updated expiry date!

Reply:
My question here is why the expiration date is set differently for a public link. Isn't the time must be set to `14:00:00` instead of `23:59:59`?

Reply:
It is expected.

from https://matrix.to/#/!QxIasTYvslDCVqoPzC:matrix.org/$fBktWHnp3ed_mdEokzmlHXkoHReuO3F5MpLpMWrnLDk?via=matrix.org&via=openproject.org&via=matrix.datenschmutz.space
>Niraj Acharya(Jankari Tech): when i create a share link for a file and set the expiration date to some thing like Friday, December 15, 2023 12:00:00 AM the experation date is set to Friday, December 15, 2023 11:59:59 PM. does the createLink endpoint set the expiration date to the last second of the day and not the given date?
https://owncloud.dev/libre-graph-api/#/drives.permissions/CreateLink
>
> Michael Barz: Yes that is correct. The OCS was behaving exactly like that.